Shared/Scripts/CopyCMakeDeps.bat

```diff
@@ -0,0 +1,353 @@
+::     If the environment variable %1 is set, we try to copy its contents
+::     to %3. If not, we try to copy the folder containing the executable %2
+::     (calling CopyParentOfExecutable).
+::     
+:: Arguments:   %1 should hold the name of the environment variable (not the value but the name)   
+::              %2 should hold the executable, this will be and argument to where
+::              %3 is the destination directory for the files
+::              %4 is a display name for the directory (in case of error) 
+::=======================================================================================================
+setlocal EnableDelayedExpansion
+set LOCATION_VARIABLE=%1
+set EXECUTABLE="%~2"
+set DESTINATION="%~3"
+set DESCRIPTION=%4
+
+if defined %LOCATION_VARIABLE% (
+    call :Verbose "     Will copy the directory for %DESCRIPTION% from the user-set environment variable %LOCATION_VARIABLE%"
+    call :CopyAllContents %DESTINATION% "%%%LOCATION_VARIABLE%%%"
+) else (
+    call :CopyParentOfExecutable  %EXECUTABLE% %DESTINATION% %DESCRIPTION%
+    if !ERRORLEVEL! NEQ 0 (
+        call :ReportNotFound %EXECUTABLE% %LOCATION_VARIABLE% %DESCRIPTION%
+    )
+)
+goto :eof
+endlocal
+
+
+::=======================================================================================================
+::====================================== SUBROUTINES ====================================================
+::=======================================================================================================
+
+:CopyFileFromEnvironmentVariableOrFromWhere
+::=======================================================================================================
+:: Subroutine: CopyFileFromEnvironmentVariableOrFromWhere
+::     If the environment variable %1 is set, we try to copy its contents (a file)
+::     to %3. If not, we try to copy the the executable specified in %2
+::     
+:: Arguments:   %1 should hold the name of the environment variable (not the value but the name)   
+::              %2 should hold the executable
+::              %3 is the destination directory for the files
+::              %4 is a display name for the directory (in case of error) 
+::=======================================================================================================
+setlocal EnableDelayedExpansion
+set LOCATION_VARIABLE=%1
+set EXECUTABLE="%~2"
+set DESTINATION="%~3"
+set DESCRIPTION=%4
+
+if defined %LOCATION_VARIABLE% (
+    call :Verbose "Will copy the directory for %DESCRIPTION% from the user-set environment variable %LOCATION_VARIABLE%"
+    set SOURCE="%%%LOCATION_VARIABLE%%%"
+) else (
+    call where /Q %EXECUTABLE%
+    if !ERRORLEVEL! NEQ 0 (
+        call :ReportNotFoundExecutable %EXECUTABLE% %LOCATION_VARIABLE% %DESCRIPTION%
+        exit /b 1
+    )
+    for /f "usebackq delims=" %%i in (`where /F "%EXECUTABLE%"`) do set SOURCE="%%~i"
+)
+XCOPY /E /D /y /q %SOURCE% %DESTINATION% > NUL 

```

Comment:
>XCOPY  [](start = 0, length = 6)

There is also an option to use robocopy... That one has a /MIR option and if the folders are large you can pass /MT to get multi-threaded copy going.